samples/detachSubscription.js

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * This sample demonstrates how to detach subscriptions with the
+ * Google Cloud Pub/Sub API.
+ *
+ * For more information, see the README.md under /pubsub and the documentation
+ * at https://cloud.google.com/pubsub/docs.
+ *
+ * Note: This sample is NOT currently included in the system-test, because
+ * the feature isn't enabled on the production server.
+ */
+
+'use strict';
+
+// sample-metadata:
+//   title: Detach Subscription
+//   description: Detaches a subscription from a topic.
+//   usage: node detachSubscription.js <existing-subscription-name>
+
+function main(subscriptionName = 'YOUR_EXISTING_SUBSCRIPTION_NAME') {
+  // [START pubsub_detach_subscription]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const subscriptionName = 'YOUR_EXISTING_SUBSCRIPTION_NAME';
+
+  // Imports the Google Cloud client library
+  const {PubSub} = require('@google-cloud/pubsub');
+
+  // Creates a client; cache this for further use
+  const pubSubClient = new PubSub();
+
+  async function detachSubscription() {
+    // Gets the status of the existing subscription
+    const sub = pubSubClient.subscription(subscriptionName);
+    const [detached] = await sub.detached();
+    console.log(
+      `Subscription ${subscriptionName} 'before' detached status: ${detached}`
+    );
+
+    await pubSubClient.detachSubscription(subscriptionName);
```

Comment:
Does `detachSubscription` throw explicit errors?  If so, we could list them in the sample.

Reply:
Do you mean things like "the subscription was already detached"? I'm actually not sure.

All of those gRPC calls are theoretically capable of throwing a bunch of gRPC errors, and the other samples don't call those out, so I'd want to keep everything as harmonized across samples there as I can.

Reply:
I was referring to explicit exceptions. Java/Python samples are generally against generic exceptions so I was wondering about Node.

```python
    try:
        publisher_client.detach_subscription(subscription_path)
    except (GoogleAPICallError, RetryError, ValueError, Exception) as err:
        print(err)
```

Reply:
Ah, I see. Yeah, the samples are all written in JavaScript, rather than TypeScript, so there's basically no rigor anywhere in them, outside of catching exceptions. I don't think even TypeScript has any real concept of typed exceptions. You just basically have to catch all the exceptions and filter for what you want. We could put a comment on it showing what exceptions might happen, but it would be very hard to figure that out for sure, and I'm afraid of it getting out of sync (e.g. if gRPC changed their exceptions thrown, and the Pub/Sub samples don't get updated).

Reply:
Thanks for explaining. I see now.